common/transport/http/src/http.ts

```diff
@@ -67,8 +67,14 @@ export class Http {
     };
 
     // Codes_SRS_NODE_HTTP_13_004: [ Use the request object from the https module when dealing with TCP based HTTP requests. ]
-    let request = https_request;
+    let request = this.useTls() ? https_request : http_request;
     if (typeof(host) === 'string') {
+      if (host.indexOf(':') !== -1) {
+        let port: string;
+        [host, port] = host.split(':');
+        httpOptions.port = parseInt(port);
+      }
```

Comment:
This port parsing stuff needs to be made robust.

Reply:
and I think our HTTP layer needs an overhaul... 

---
In reply to: [195307110](https://github.com/Azure/azure-iot-sdk-node/pull/315#discussion_r195307110) [](ancestors = 195307110)

Reply:
on second thought: can't we use the setOptions call to pass the port? 

---
In reply to: [195512367](https://github.com/Azure/azure-iot-sdk-node/pull/315#discussion_r195512367) [](ancestors = 195512367,195307110)